src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -498,12 +498,15 @@ class NormalSchedulingQueue : public SchedulingQueue {
  public:
   NormalSchedulingQueue(){};
 
-  bool TaskQueueEmpty() const { return pending_normal_tasks_.empty(); }
+  bool TaskQueueEmpty() const EXCLUSIVE_LOCKS_REQUIRED(mu_) {
+    return pending_normal_tasks_.empty();
```

Comment:
Since this is a public method, but the mutex is private, I think you need to hold the lock here instead of marking it as `EXCLUSIVE_LOCKS_REQUIRED`. The places where you call this internally that are already holding the lock could just call `pending_normal_tasks_.empty()` directly.

Reply:
Sounds good! If I acquire a lock within the TaskQueueEmpty() method, it looks like I need to remove the `const` keyword from the method (otherwise it won't compile). Is that ok?

Reply:
Oh yes, you can avoid that by marking the mutex field as `mutable`. That tells the compiler that it's okay to mutate the `mu_` variable in const methods.

Reply:
Oh cool! I'll do that